### PR TITLE
[bitnami/keycloak] Add custom certs to system truststore

### PR DESCRIPTION
**Pull request by moritzwiechers**
to enable https-communication with hosts with certificates generated by custom ca's.

### Description of the change

Added the option to specify a secret which contains the custom ca certificates. They will be mounted to the container and loaded on keycloak startup by setting the environment variable KC_TRUSTSTORE_PATHS

### Benefits

- Easy way to add custom-ca's to keycloak to communicate with onprem ldap/identity provider etc.
- No need to configure keystores/truststore, add keys or other things related to the tls settings

### Applicable issues
Should resolve:
- [https://github.com/bitnami/charts/issues/6602](https://github.com/bitnami/charts/issues/6602)
- [https://github.com/bitnami/charts/issues/7536](https://github.com/bitnami/charts/issues/7536)
- [https://github.com/bitnami/charts/issues/7796](https://github.com/bitnami/charts/issues/7796)
- [https://github.com/bitnami/charts/issues/10829](https://github.com/bitnami/charts/issues/10829)

### Checklist

- [X] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/). This is *not necessary* when the changes only affect README.md files.
- [X] Variables are documented in the values.yaml and added to the `README.md` using [readme-generator-for-helm](https://github.com/bitnami/readme-generator-for-helm)
- [X] Title of the pull request follows this pattern [bitnami/<name_of_the_chart>] Descriptive title
- [X] All commits signed off and in agreement of [Developer Certificate of Origin (DCO)](https://github.com/bitnami/charts/blob/main/CONTRIBUTING.md#sign-your-work)

### CODE REVIEW

**Review by migruiz4 — COMMENTED** [review]
Hi @moritzwiechers,

Thank you for your contribution! Since this is a new feature, could you please bump a minor version?

**Review by migruiz4 — COMMENTED** [review]
Could you please update the value description? I left a suggestion below:

File: bitnami/keycloak/values.yaml [hunk 1]
```diff
@@ -138,6 +138,11 @@ auth:
   ## @param auth.annotations Additional custom annotations for Keycloak auth secret object
   ##
   annotations: {}
+## Custom Certificates
+## @param customCaExistingSecret define a secret for merging certificates to the system trustore like custom ca-certs (pem, pkcs12)
```
[the remark this answers is not in the document]
Comment (moritzwiechers) [reply bitnami/keycloak/values.yaml]: Updated the value description. Put it in one commit with readme update version bump

**Comment by moritzwiechers** [comment]
@migruiz4 Is there anything you need from me to advance with your review?

**Review by migruiz4 — APPROVED** [review]
Looks good to me!

**Comment by luhahn** [comment]
thats basically what were currently trying to do, does it work for you this way? It still seems ldaps ignores our certificates.

**Comment by moritzwiechers** [comment]
> thats basically what were currently trying to do, does it work for you this way? It still seems ldaps ignores our certificates.

Hey,
It should work for ldap, the docs name it outgoing ssl connections. I tested it with an ms adfs configured as oidc identity Provider. Maybe you could show your config. And the start up log from your keycloak container.
Also please make sure, that the server you a trying to connect sends the correct cert including the whole certificate chain. (especially when you use intermediate ca's [explanation](https://knowledge.digicert.com/solution/how-certificate-chains-work)

**Comment by chee-yee** [comment]
Hi, please can someone share an example of the required configuration in the YAML file for adding a certificate which has been created as a secret named "keycloak-ldaps-cert"?

I've added the below but still LDAPS is not able to connect - do I need to specify additional config in order for the certificate to be referenced by KeyCloak?

customCaExistingSecret: keycloak-ldaps-cert

**Comment by moritzwiechers** [comment]
hey,
adding
customCaExistingSecret: keycloak-ldaps-cert 

is the right way. You need to create the secret before and it has to be in the same namespace, e.g.:

kubectl --namespace <keycloak-namespace> create secret generic keycloak-ldaps-cert --from-file=<path/to/your/.keycloak-ldaps-cert.pem>

Maybe you should check your keycloak log for any errors. When successfully loaded a message like this should appear on startup:

keycloak 2024-08-15 13:09:47,763 INFO  [org.keycloak.truststore.TruststoreBuilder] (main) Found the following truststore files under directories specified in the truststore paths [/opt/bitnami/keycloak/custom-ca/keycloak-ldaps-cert.pem, /opt/bitnami/keycloak/custom-ca/..data/keycloak-ldaps-cert.pem, /opt/bitnami/keycloak/custom-ca/..2024_08_15_13_08_42.3315499878/keycloak-ldaps-cert.pem

**Comment by chee-yee** [comment]
Thanks for responding @moritzwiechers and will give it a try. Do you know if we need to specify any other parameters in the yaml file for KeyCloak to reference this certificate during LDAPS connection or is this the only config required?

**Pull request merged**